.circleci/config.yml

```diff
@@ -292,94 +294,85 @@ jobs:
               parallelism:<< parameters.parallelism >> \
               arch:<< parameters.arch >> \
             "
-
-            echo "some combinations may run nothing"
-      - unless:
-          # This works around circleci limitation by skipping tests for combinations that don't make sense
-          # See: https://discuss.circleci.com/t/matrix-exclude-entire-subset/43879
-          condition:
-            or:
-              - and:
-                  # On PRs only run some base tests (only 1 k8s version, no ipv6, no arm64)
-                  - or:
-                      - not: {matches: {pattern: "^release-.*", value: << pipeline.git.branch >>}}
-                      - not: {equal: [master, << pipeline.git.branch >>]}
-                  - or:
-                      - << parameters.ipv6 >>
-                      - equal: ["v1.19.16-k3s1", << parameters.k8sVersion >>]
-                      - equal: ["arm64", << parameters.arch >>]
-              - and:
-                # Ipv6 only works with kind
-                - << parameters.ipv6 >>
-                - not: {equal: [kind, << parameters.k8sVersion >>]}
-              - and:
-                # disallow running tests on kind except if it's ipv6 or it's universal
-                - equal: [kind, << parameters.k8sVersion >> ]
-                - not:
-                    or:
-                      - << parameters.ipv6 >>
-                      - equal: [test/e2e-universal, << parameters.target >>]
+            function skip() {
+              echo "Non valid job combination halting job reason: $1"
+              circleci-agent step halt
+              exit 0
+            }
+            if [[ "<< parameters.arch >>" == "arm64" && "<< parameters.ipv6 >>" == "true" ]]; then
+              skip "don't run v6 tests on arm64"
+            fi
+            if [[ "<< parameters.k8sVersion >>" == "kind" ]]; then
+              if [[ "<< parameters.ipv6 >>" != "true" && "<< parameters.target >>" != "test/e2e-universal" ]]; then
+                skip "kind should only be used when testing ipv6 or with test/e2e-universal"
+              fi
+            fi
+            if [[ "<< pipeline.git.branch >>" != "master" && "<< pipeline.git.branch >>" != "release-*" ]]; then
+              if [[ "<< parameters.ipv6 >>" == "true" || "<< parameters.k8sVersion >>" == "v1.19.16-k3s1" || "<< parameters.arch >>" == "arm64" ]]; then
+                skip "Not running tests on PRs with ipv6, oldK8s and arm64"
+              fi
+            fi
+            echo "Continuing tests"
```

Comment:
This diff is harder to read than it should. Whatever is after this is just indent change